### PR TITLE
[Objective-C API] Enable CoreML EP

### PR DESCRIPTION
**Pull request by edgchen1**
**Description**
Enable CoreML EP in Objective-C API.

**Motivation and Context**
Add functionality to API.

### CODE REVIEW

File: objectivec/src/ort_coreml_execution_provider.mm [hunk 1]
```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import "ort_coreml_execution_provider.h"
+
+#if __has_include("coreml_provider_factory.h")
```
Comment (gwang-msft) [inline objectivec/src/ort_coreml_execution_provider.mm:6]: Can we use #ifdef USE_COREML here?

Comment (edgchen1) [reply objectivec/src/ort_coreml_execution_provider.mm:6]: I don't think so - in the pod, this will build separately from ORT

**Review by gwang-msft — DISMISSED** [review]

File: objectivec/test/ort_session_test.mm [hunk 2]
```diff
@@ -191,6 +192,28 @@ - (void)testRunFailsWithInvalidInput {
...
 
+- (void)testAppendCoreMLEP {
+  NSError* err = nil;
+  ORTSessionOptions* sessionOptions = [ORTSessionTest makeSessionOptions];
+  ORTCoreMLExecutionProviderOptions* coreMLOptions = [[ORTCoreMLExecutionProviderOptions alloc] init];
+
+  BOOL appendResult = [sessionOptions appendCoreMLExecutionProviderWithOptions:coreMLOptions
+                                                                         error:&err];
+
+  if (!ORTIsCoreMLExecutionProviderAvailable()) {
```
Comment (skottmckay) [inline objectivec/test/ort_session_test.mm:203]: Is there a test where we set one of the options?

Comment (edgchen1) [reply objectivec/test/ort_session_test.mm:203]: I could set some option values - not sure how to check the effects of doing that though. Open to suggestions for this.

**Review by gwang-msft — APPROVED** [review]

File: objectivec/test/ort_session_test.mm (hunk 2 above)
Comment (skottmckay) [reply objectivec/test/ort_session_test.mm:204]: Probably fine to manually validate in a debugger for now if there's no better way. 

Just wanted to make sure things work as expected and that there was example code of actually setting an option.

**Review by skottmckay — APPROVED** [review]

**Pull request merged**